purchase_advance_payment/tests/test_purchase_advance_payment.py

```diff
@@ -146,6 +146,8 @@ def setUpClass(cls):
                 "currency_id": cls.currency_usd.id,
             }
         )
+        if cls.purchase_order_1.currency_id == cls.purchase_order_1.company_id.currency_id:
+            cls.purchase_order_1.currency_id = cls.currency_usd
```

Comment:
Shouldn't we then always force USD currency? Or it will not work correctly with EUR then? Also would be nice to have a comment explaining why this is needed.

Either way, check seems strange. Maybe just check what kind of currency company currency is and decide. Why order currency must match company currency to switch to usd?..

Reply:
By the way, can you enable pre-commit on this submodule and run it, so it would fix code according to OCA guidelines.

Reply:
Before our change, company and order already had different currencies. And these tests only works when currencies are different.